### PR TITLE
feat: custom conflict resolution implementation in Swift

### PR DESCRIPTION
**Pull request by jayahariv**
* these files are not referenced in the project, since the previous commit is not yet merged.

Ref: #2427

### CODE REVIEW

**Comment by jayahariv** [comment]
i kept this PR without assigning you, also not added any files to project too. Since the objc code is still not merged. Will finish implementation once the it is merged.

File: Swift/ReplicatorConfiguration.swift [hunk 1]
```diff
@@ -225,3 +237,18 @@ public class ReplicatorConfiguration {
         }
     }
 }
+
+
+/* internal */ fileprivate class CustomCBLConflictResolver: NSObject, CBLConflictResolver {
```
Comment (jayahariv) [inline Swift/ReplicatorConfiguration.swift]:
```suggestion
/* internal */ fileprivate class BridgingConflictResolver: NSObject, CBLConflictResolver {
```

File: Swift/ReplicatorConfiguration.swift [hunk 2]
```diff
@@ -209,6 +220,7 @@ public class ReplicatorConfiguration {
         c.documentIDs = self.documentIDs
         c.pushFilter = self.filter(push: true)
         c.pullFilter = self.filter(push: false)
+        c.conflictResolver = self.conflictResolver != nil ? CustomCBLConflictResolver(self.conflictResolver!) : nil
```
Comment (jayahariv) [inline Swift/ReplicatorConfiguration.swift]:
```suggestion
        c.conflictResolver = self.conflictResolver != nil ? BridgingConflictResolver(self.conflictResolver!) : nil
```

File: Swift/Conflict.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    /// The document which is already in the database.
+    public var localDocument: Document? {
+        guard let doc = impl.localDocument else {
+            return nil
+        }
+        return Document(doc)
+    }
+    
+    /// The document which is merging to the database.
+    public var remoteDocument: Document? {
```
Comment (pasin) [inline Swift/Conflict.swift]: Change comment to  `The document replicated from the remote database.`.

File: Swift/Conflict.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Conflict class
+public struct Conflict {
+    
+    /// The document which is already in the database.
+    public var localDocument: Document? {
```
Comment (pasin) [inline Swift/Conflict.swift]: Change comment to `The document in the local database`.

File: Swift/ConflictResolution.swift [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Conflict Resolution class
+public class ConflictResolution {
+    
+    /// The default conflict resolver object.
+    public static var `default`: ConflictResolver {
```
Comment (pasin) [inline Swift/ConflictResolution.swift]: Change the comment to the `default conflict resolver used by the replicator`.

File: Swift/Conflict.swift (hunk 4 above)
Comment (pasin) [reply Swift/Conflict.swift]: We will need to make sure to ObjC as well.

File: Swift/Conflict.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Conflict class
+public struct Conflict {
+    
+    /// The document which is already in the database.
```
Comment (jayahariv) [inline Swift/Conflict.swift]:
```suggestion
    /// The document in the local database.
```

File: Swift/Conflict.swift [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    public var localDocument: Document? {
+        guard let doc = impl.localDocument else {
+            return nil
+        }
+        return Document(doc)
+    }
+    
+    /// The document which is merging to the database.
```
Comment (jayahariv) [inline Swift/Conflict.swift]:
```suggestion
    /// The document replicated from the remote database
```

File: Swift/ConflictResolution.swift [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Conflict Resolution class
+public class ConflictResolution {
+    
+    /// The default conflict resolver object.
```
Comment (jayahariv) [inline Swift/ConflictResolution.swift]:
```suggestion
    /// The default conflict resolver used by the replicator.
```

File: Swift/Conflict.swift (hunk 4 above)
Comment (jayahariv) [reply Swift/Conflict.swift]: what is the document is deleted? should we include a line saying, `If returns nil, document is deleted.`

**Pull request merged**